spec/presenters/tree_builder_ops_rbac_spec.rb

```diff
@@ -83,4 +83,16 @@ def x_get_tree_custom_kids_for_and_expect_objects(type_of_model, expected_object
       end
     end
   end
+
+  describe "#x_get_tree_tenant_kids" do
+    let!(:child_tenant) { FactoryGirl.create(:tenant, :parent => Tenant.root_tenant) }
+    let!(:child_tenant_out_of_region) { FactoryGirl.create(:tenant, :parent => Tenant.root_tenant, :id => Tenant.root_tenant.id * 3) }
```

Comment:
@ZitaNemeckova 

you created child tenant in default region and his child which is "probably"( * 3) in different region and this situation doesn't happen in real.

we need to create tenant structure in default region and then whole tenant structure in other tenant region

and then check whether it is listing only tenant in current region.

you can get some inspiration from https://github.com/ManageIQ/manageiq/pull/17453/files#diff-de9adf0bd8114667190fecb643922d79R941 .

Reply:
Using`id_in_region` sounds better 👍